### ISSUE TITLE
Implement properties panel extension as a provider

### ISSUE THREAD

**Issue by nikku** [camunda/camunda-modeler#2133]
__What should we do?__

Building upon https://github.com/bpmn-io/bpmn-js-properties-panel/pull/438 we should implement our [properties panel extension](https://github.com/camunda/camunda-modeler/blob/develop/client/src/plugins/element-templates-modal/modeler/PropertiesProvider.js) as a provider.

__Why should we do it?__

In order to address https://github.com/camunda/camunda-modeler/issues/1992

**Comment by nikku** [comment]
Depends on properties panel and `camunda-bpmn-js` release.

**Comment by pinussilvestrus** [comment]
An interesting side effect I encountered with the _current_ implementation to extend the existing properties provider instead of registering a new one

* the BpmnEditor uses the PropertiesPanel version coming from `camunda-bpmn-js`
* the Element Templates Modal plugin uses the PropertiesPanel (provider) version [specified in `client`
](https://github.com/camunda/camunda-modeler/blob/develop/client/src/plugins/element-templates-modal/modeler/PropertiesProvider.js#L11)
--> There might be a version mismatch, which can lead to side effects, especially when the properties provider is assuming some functionality inside the Properties Panel module, which might be not there, in a different version.

tldr: it's great that we're going to update the plugin to act as a provider, and not extending anything.

**Comment by nikku** [comment]
The problematic usage is the one way how users currently extend the properties panel by shipping their own providers, right?

I believe it is best to simply let that way of providing the properties panel die.

Alternatively we could make `bpmn-js-properties-panel` a peer dependency in `camunda-bpmn-js`. This way we can ensure that the properties panel is always shared across camunda-bpmn-js _and_ providers. Any ideas?

**Comment by pinussilvestrus** [comment]
> I believe it is best to simply let that way of providing the properties panel die.

I'd rather go with this. I'd like to avoid adding peer-dependencies to `camunda-bpmn-js` since the actual use case is that you don't have to install extra dependencies to make it run.
